x-pack/plugins/alerting/server/rules_client/methods/update.ts

```diff
@@ -241,12 +257,16 @@ async function updateAlert<Params extends RuleTypeParams>(
     );
   }
 
+  const allActions = [...data.actions, ...(data.systemActions ?? [])];
   // Extract saved object references for this rule
-  const {
-    references,
-    params: updatedParams,
-    actions,
-  } = await extractReferences(context, ruleType, data.actions, validatedAlertTypeParams);
+  const { references, params: updatedParams } = await extractReferences(
```

Comment:
Same here: `extractReferences` calls `denormalizedActions` and returns `actionsWithRefs`.